src/heap/new-spaces.cc

```diff
@@ -444,6 +475,134 @@ SemiSpaceNewSpace::SemiSpaceNewSpace(Heap* heap,
   DCHECK_LE(target_capacity_, maximum_capacity_);
 }
 
+#ifdef V8_COMPRESS_POINTERS_IN_MULTIPLE_CAGES
+SemiSpaceNewSpace::SemiSpaceNewSpace(Heap* heap,
+                                     SemiSpaceNewSpace* original_space)
+    : NewSpace(heap, original_space),
+      to_space_(heap, kToSpace),
+      from_space_(heap, kFromSpace),
+      minimum_capacity_(original_space->minimum_capacity_),
+      maximum_capacity_(original_space->maximum_capacity_),
+      target_capacity_(original_space->target_capacity_) {
+  // Recreate remapped clones of the original pages.
```

Comment:
We should consider requiring a that newspace be empty before cloning.  Doesn't necessarily imply full GC, ISTR there is an invariant that two minor GCs will empty the nursery, but it would be a win if we did not have to maintain this code.